src/validation/JoiValidator.js

```diff
@@ -0,0 +1,40 @@
+import Joi from 'joi';
+import validationRules from './validationRules';
+import messages from '../utils/messages';
+
+/** *
+ *  class that help to validate user details
+ */
+const JoiValidation = {
+
+  validateName(name) {
```

Comment:
You can create a generic function that will create the regex schema to replace the functions `validateName`, and `validateEmail`. The argument for the generic function will be the validation rule (that is the regex rule).